errai-bus/src/main/java/org/jboss/errai/bus/server/service/ErraiServiceProxy.java

```diff
@@ -28,10 +28,14 @@
  */
 class ErraiServiceProxy implements ErraiService<Object> {
 
-  private MessageBusProxy messageBusProxy = new MessageBusProxy();
-  private RequestDispatcherProxy requestDispatcherProxy = new RequestDispatcherProxy();
+  private MessageBusProxy messageBusProxy;
+  private RequestDispatcherProxy requestDispatcherProxy;
   private ErraiService service;
 
+  public ErraiServiceProxy() {
+    reset();
```

Comment:
I think you will need different logic for initialising and resetting. Please see comment on reset.